src/libraries/System.Data.OleDb/src/DbPropSet.COMWrappers.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Data.Common;
+using System.Diagnostics;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Data.OleDb
+{
+    internal sealed partial class DBPropSet
+    {
+        private unsafe void SetLastErrorInfo(OleDbHResult lastErrorHr)
+        {
+            // note: OleDbHResult is actually a simple wrapper over HRESULT with OLEDB-specific codes
+            string message = string.Empty;
+            IntPtr pErrorInfo;
+            OleDbHResult errorInfoHr = UnsafeNativeMethods.GetErrorInfo(0, &pErrorInfo);  // 0 - IErrorInfo exists, 1 - no IErrorInfo
+            if ((errorInfoHr == OleDbHResult.S_OK) && (pErrorInfo != IntPtr.Zero))
+            {
+                UnsafeNativeMethods.IErrorInfo errorInfo = (UnsafeNativeMethods.IErrorInfo)OleDbComWrappers.Instance
+                    .GetOrCreateObjectForComInstance(pErrorInfo, CreateObjectFlags.UniqueInstance);;
+                try
+                {
+                    ODB.GetErrorDescription(errorInfo, lastErrorHr, out message);
+                    // note that either GetErrorInfo or GetErrorDescription might fail in which case we will have only the HRESULT value in exception message
+                }
+                finally
+                {
+                    if (errorInfo is IDisposable disposable)
```

Comment:
This is the same bug that was in WinForms, right? The `ErrorInfoWrapper` has a `Dispose` method, but doesn't implement IDisposable.

Another good reason to do a straight cast here, and not use `is/as` when you expect an object to implement an interface.